[CS2113-T18-3] Budget_planner

bljhty opened the pull request:
A financial tracker for general users

irving11119 commented on src/main/java/seedu/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,227 @@
+ * This command supports searching by a specific date or by a month. If a date isn't provided, it will not be used
+ * as a filter. Similarly, if the category or description isn't provided, they won't be used as filters.
+ */
+public class FindCommand extends Commands {
```
Good use of inheritance

irving11119 commented on src/main/java/seedu/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,227 @@
+                    throw new KaChinnnngException("Failed to create directory " + dir.getAbsolutePath());
+                }
+            }
+            FileHandler fh = new FileHandler("logs/FindCommand.log", true);
```
You could perhaps change `logs/FindCommand.log` into a static variable

irving11119 commented on src/main/java/seedu/duke/financialrecords/FinancialRecord.java:
```diff
@@ -0,0 +1,152 @@
+     * @throws KaChinnnngException if there is an error in the command
+     */
+    public void setAmount(double amount) throws KaChinnnngException {
+        assert amount >= 0 : "amount should not be negative";
```
Good use of assertions in this class

irving11119 commented on src/main/java/seedu/duke/parser/ExpenseParser.java:
```diff
@@ -0,0 +1,120 @@
+            if (expenseTypeString.equals("breakfast")) {
+                return new Food(expenseDescriptionString, expenseDate, expenseAmount, MealType.BREAKFAST);
+            } else if (expenseTypeString.equals("lunch")) {
+                return new Food(expenseDescriptionString, expenseDate, expenseAmount, MealType.LUNCH);
+            } else if (expenseTypeString.equals("dinner")) {
+                return new Food(expenseDescriptionString, expenseDate, expenseAmount, MealType.DINNER);
+            } else {
+                return new Food(expenseDescriptionString, expenseDate, expenseAmount, MealType.UNDEFINED);
+            }
+        } else if (expenseCategoryString.equals("transport")){
+            if (expenseTypeString.equals("train")) {
+                return new Transport(expenseDescriptionString, expenseDate, expenseAmount,
+                        TransportationType.TRAIN);
+            } else if (expenseTypeString.equals("bus")) {
+                return new Transport(expenseDescriptionString, expenseDate, expenseAmount,
+                        TransportationType.BUS);
+            } else if (expenseTypeString.equals("taxi")) {
+                return new Transport(expenseDescriptionString, expenseDate, expenseAmount,
+                        TransportationType.TAXI);
+            } else if (expenseTypeString.equals("fuel")) {
+                return new Transport(expenseDescriptionString, expenseDate, expenseAmount,
+                        TransportationType.FUEL);
+            } else {
+                return new Transport(expenseDescriptionString, expenseDate, expenseAmount,
+                        TransportationType.UNDEFINED);
+            }
+        } else if (expenseCategoryString.equals("utilities")) {
+            if (expenseTypeString.equals("water")) {
+                return new Utilities(expenseDescriptionString, expenseDate, expenseAmount, UtilityType.WATER);
+            } else if (expenseTypeString.equals("electricity")) {
+                return new Utilities(expenseDescriptionString, expenseDate, expenseAmount, UtilityType.ELECTRICITY);
+            } else if (expenseTypeString.equals("gas")) {
+                return new Utilities(expenseDescriptionString, expenseDate, expenseAmount, UtilityType.GAS);
+            } else{
+                return new Utilities(expenseDescriptionString, expenseDate, expenseAmount, UtilityType.UNDEFINED);
+            }
+        } else {
+            throw new KaChinnnngException("Please enter a valid category");
+        }
+    }
```
You could look at reducing nesting here to reduce arrowhead code!

irving11119 commented on src/main/java/seedu/duke/commands/ListCommand.java:
```diff
@@ -0,0 +1,73 @@
+            return;
+        }
+        // Print incomes
+        if (incomes.isEmpty()) {
+            System.out.println("You have no recorded incomes.");
+        } else {
+            System.out.println("Here are your incomes:");
+            for (int i = 0; i < incomes.size(); i++) {
+                System.out.println((i + 1) + ". " + incomes.get(i).toString());
+                totalIncome += incomes.get(i).getAmount();
+            }
+            System.out.printf("Total income is: $%.2f.\n", totalIncome);
```
Personally I feel you could abstract this out into a method as it is quite similar to your for loop below. This also helps to make your code neater.

irving11119 commented on src/main/java/seedu/duke/commands/UsageInstructions.java:
```diff
@@ -0,0 +1,37 @@
+    /**
+     * Displays usage instructions for the application, including command formats and a link to the user guide.
+     */
+    public void getHelp(){
+        ui.showLineDivider();
+        // Shows a message linking the user to the user guide of our app
+        System.out.println("You can access our user guide by " +
+                "https://docs.google.com/document/d/1BOz_v4eYQ8y7Dje6Jm6nqymi9jmrsb9MAohLCL_sLvI/edit?usp=sharing\n");
+        // Displays the various commands that user can use including the respective formats of the commands
+        System.out.println("Functions and their format:\n");
+        System.out.println("Adding an entry: add\nFormat:\n" + "Add expense /category /description /value\n" +
+                "Add income /description /value\n");
+        System.out.println("Listing all entries: list\nFormat:\n" + "list\n" + "list income\n" +
+                "list expense\n");
+        System.out.println("Deleting an entry: delete\nFormat:\ndelete income [index_pos]\n" +
+                "delete expense [index_pos]\n");
+        System.out.println("Check balance of income: balance\nFormat:\n" + "balance\n");
+        System.out.println("Exiting the program: exit\nFormat:\n" + "exit");
+        ui.showLineDivider();
```
You could consider constants for this rather than string literals. You could make your entire menu in a string constant and thus simplify this method.

irving11119 commented on src/main/java/seedu/duke/storage/GetFromTxt.java:
```diff
@@ -0,0 +1,118 @@
+                double amountFromFile = Double. parseDouble(textLine.split(" \\| ")[2]);
+                LocalDate dateFromFile =  LocalDate.parse(textLine.split(" \\| ")[3]);
+                // Check for valid amount range
+                if (amountFromFile > 999999.99 || amountFromFile <= 0) {
```
Perhaps you could convert this into a constant variable rather than a string literal in case you'd like to change your boundary values in the future and it needs to be used elsewhere in your codebase.

irving11119 reviewed (commented):
Overall, I think codebase is well abstracted into different classes and has good OOP practices. Just look at ways to make parts of your code more readable and succinct as it makes your code more readable and neat.

bljhty committed:
Merge pull request #114 from bljhty/Bryan-FindParserTest

Added JUnit tests and javadocs

KenCheung18 committed:
Add JUnit test for clear function

KenCheung18 committed:
JUnit test for all clear functions

KenCheung18 committed:
JavaDoc comments for tests

bljhty committed:
refactored main class, added JUnit test for dukeclass and DukeTest

bljhty committed:
ediited code

bljhty committed:
edited expected and actual results

KenCheung18 committed:
JUnit test for saveToTextFile

bljhty committed:
no message

bljhty committed:
no message

KenCheung18 committed:
Save to txt test bug fix

bljhty committed:
no message

bljhty committed:
no message

bljhty committed:
no message

bljhty committed:
no message

KenCheung18 committed:
JUnit test for getFromFTxt

bljhty committed:
Merge pull request #116 from bljhty/Bryan-moreJUnitTest

Bryan more j unit test

KenCheung18 committed:
Bug fix on JUnit test for clear function

KenCheung18 committed:
Change back

KenCheung18 committed:
Merge branch 'master' into JUnit-test-for-clear-function

KenCheung18 committed:
Changes to the JUnit test

KenCheung18 committed:
Bug Fix

KenCheung18 committed:
Merge branch 'JUnit-test-for-clear-function' of https://github.com/KenCheung18/tp into JUnit-test-for-clear-function

KenCheung18 committed:
Merge pull request #117 from KenCheung18/JUnit-test-for-Storage

JUnit test for storage

KenCheung18 committed:
Bug fix

KenCheung18 committed:
Merge pull request #115 from KenCheung18/JUnit-test-for-clear-function

J unit test for clear function

Hongzhii committed:
add expenseManager unit tests

Hongzhii committed:
Merge branch 'add-expense-manager-unit-tests'

* add-expense-manager-unit-tests:
  add expenseManager unit tests

Hongzhii committed:
Merge pull request #118 from Hongzhii/master

add expenseManager unit tests

limyuhching committed:
Add ExchangeRateFileHandler

ExchangeRateFileHandler is for loading/saving exchange rates from /data/ExchangeRates.txt to the ExchangeRateManager

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/tp into Yuh.Ching-feature-currency.conversion

# Conflicts:
#	src/main/java/seedu/duke/Duke.java
#	src/main/java/seedu/duke/commands/FindCommand.java

limyuhching committed:
Add ExchangeRateManagerTest

limyuhching committed:
Add ExchangeRateFileHandlerTest

limyuhching committed:
Update runtest.sh

limyuhching committed:
Update .gitignore

limyuhching committed:
Merge pull request #119 from limyuhching/Yuh.Ching-feature-currency.conversion

Yuh.ching feature currency.conversion

json-bot-boop commented on docs/DeveloperGuide.md:
```diff
@@ -7,7 +7,57 @@
+3. `IncomeManager` then adds the `Income` object to the `IncomeList`.
+4. `IncomeManager` then returns a `String` to `Ui` to be printed out to the user.
+
+![incomeManager_sequence_diagram.png](..%2Fimages%2FincomeManager_sequence_diagram.png)
```
Maybe can try to fix the image path, image not loading

json-bot-boop commented on docs/DeveloperGuide.md:
```diff
@@ -7,7 +7,57 @@
+5. Expense object is returned to ExpenseManager from ExpenseParser.
+6. Expense obejct is returned to Duke from ExpenseManager.
+
+![expenseManager_sequence_diagram.png](/Users/apple/Desktop/expenseManager_sequence_diagram.png)
```
Maybe can try to fix the image path, image not loading

json-bot-boop commented on docs/DeveloperGuide.md:
```diff
@@ -7,7 +7,57 @@
+6. If users searches through the expense list, either the category or date will be used to search through the expense list.
+7. The execute method will then be called to search through the list and return the list of expenses or incomes that matches the search criteria.
+
+![FindCommand_SequenceDiagram.png](..%2Fimages%2FFindCommand_SequenceDiagram.png)
```
Maybe can try to fix the image path, image not loading

json-bot-boop commented on docs/DeveloperGuide.md:
```diff
@@ -7,7 +7,57 @@
+6. If amount in txt file exceed limit or incorrect, throw `KaChinnnningException`
+7. Read the content on the file and add corresponding object to incomes or expenses list
+
+![FindCommand_SequenceDiagram.png](..%2Fimages%2FGetFromTxt.png)
```
Maybe can try to fix the image path, image not loading

limyuhching committed:
Update UserGuide.md

Update add income and expense
Add edit function

json-bot-boop commented on images/GetFromTxt.png:
Maybe do not show the try-catch in the sequence diagram

json-bot-boop commented on images/incomeManager_sequence_diagram.png:
maybe typo for one of the function, incomAmount (From IncomeParser to Incomes:Income)

json-bot-boop commented on docs/DeveloperGuide.md:
```diff
@@ -7,7 +7,57 @@
 ## Design & implementation
```
Maybe can have a brief summary of how each component links to each other

sRanay reviewed (commented):
Overall documentation is ok but maybe need to check if the image is loaded after pushing

bljhty committed:
updated DG and edited some code

bljhty committed:
Merge pull request #120 from bljhty/Bryan-update_DG2

updated DG and edited some code

bljhty committed:
uploaded image

bljhty committed:
Merge pull request #121 from bljhty/Bryan-update_DG2

uploaded image

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Add currency conversion mechanism to IncomeParser

Update input and expected txt file to test for currency conversion when adding income

Debug:
- Removed try and catch clause at executeCommand() add income and add expense cases to update

limyuhching committed:
Merge branch 'Yuh.Ching-feature-currency.conversion' of https://github.com/limyuhching/tp into Yuh.Ching-feature-currency.conversion

limyuhching committed:
Merge branch 'master' into Yuh.Ching-feature-currency.conversion

limyuhching committed:
Update EXPECTED.TXT

limyuhching committed:
Merge pull request #122 from limyuhching/Yuh.Ching-feature-currency.conversion

Add currency conversion function to add income command

KenCheung18 committed:
Add user guide and minor change to the Usage Instructions

KenCheung18 committed:
Merge branch 'master' into User-guide-and-add-new-content-on-help-function

KenCheung18 committed:
changed the expected txt

KenCheung18 committed:
Merge branch 'User-guide-and-add-new-content-on-help-function' of https://github.com/KenCheung18/tp into User-guide-and-add-new-content-on-help-function

KenCheung18 committed:
Merge pull request #123 from KenCheung18/User-guide-and-add-new-content-on-help-function

Add user guide and minor change to the Usage Instructions

Hongzhii committed:
update DG

Hongzhii committed:
Merge branch 'update-expense-manager-uml-diagram'

* update-expense-manager-uml-diagram:
  update DG

Hongzhii committed:
Merge pull request #124 from Hongzhii/master

update DG

KenCheung18 committed:
Change to developer guide

KenCheung18 committed:
Merge pull request #125 from KenCheung18/Developer-guide-changes

Change to developer guide

Hongzhii committed:
add expense class diagram

Hongzhii committed:
Merge branch 'add-puml-expense-class-diagram'

* add-puml-expense-class-diagram:
  add expense class diagram

Hongzhii committed:
Merge pull request #126 from Hongzhii/master

add expense class diagram

bljhty committed:
no message

bljhty committed:
Merge branch 'master' into Bryan-update_UG2

bljhty committed:
Merge pull request #127 from bljhty/Bryan-update_UG2

no message

Hongzhii committed:
update expense manager uml diagram

Hongzhii committed:
Merge branch 'update-expenseManager-sequence-diagram'

* update-expenseManager-sequence-diagram:
  update expense manager uml diagram

limyuhching committed:
Update DG for ExchangeRateManager and UG for currency features

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/UserGuide.md

limyuhching committed:
Merge pull request #129 from limyuhching/master

Add DG for ExchangeRateManager and Update UG

Hongzhii committed:
Merge pull request #128 from Hongzhii/master

update expense manager uml diagram

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Update Ui output of list exchange rate

limyuhching committed:
Merge branch 'AY2324S1-CS2113-T18-3:master' into master

limyuhching committed:
Merge pull request #130 from limyuhching/master

Update Ui output of list exchange rate function

limyuhching committed:
Update DeveloperGuide.md

limyuhching committed:
Merge pull request #131 from limyuhching/master

Update DeveloperGuide.md

KenCheung18 committed:
help function update for currency update

KenCheung18 committed:
change to test-ui-test

KenCheung18 committed:
bug fix

KenCheung18 committed:
Merge pull request #132 from KenCheung18/help-function-update

Help function update

limyuhching committed:
Update error messages

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/tp

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/tp

limyuhching committed:
Merge pull request #133 from limyuhching/master

Update error messages

limyuhching committed:
Update UserGuide.md

Hongzhii committed:
no message

Hongzhii committed:
Merge branch 'fix-DG-images'

* fix-DG-images:
  no message

Hongzhii committed:
Merge pull request #134 from Hongzhii/master

fix DG images

TeoHaoZhi commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,112 @@
+
+The architectural diagram of KaChinnnng is as follows:
+
+![ArchitectureDiagram.png](..%2Fimages%2FArchitectureDiagram.png)
```
Is the diagram rendered correctly? On the webpage no image is shown?

![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110603186/861dd41b-ed3f-4ada-bb90-1fe86e7fdef4)

TeoHaoZhi commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,112 @@
+![FindCommand_SequenceDiagram.png](..%2Fimages%2FSaveToTxt.png)
 
 ## Product scope
 ### Target user profile
```
Is the information for the target user profile missing?

TeoHaoZhi commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,112 @@
+
+![UpdateExchangeRate_SequenceDiagram.png](..%2Fimages%2FUpdateExchangeRate_SequenceDiagram.png)
+
+This is facilitated by `Duke`, `UpdateExchangeRateCommand`, `ExchangeRateManager`'and `ExchangeRateFileHandler`.
```
Could the ExchangeRateManager be more descriptive to aid in understanding using the sequence diagram?

TeoHaoZhi commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,112 @@
+
+This is facilitated by `Duke`, `UpdateExchangeRateCommand`, `ExchangeRateManager`'and `ExchangeRateFileHandler`.
+
+### SaveToTxt Command
```
Could there be a sequence diagram added to this to aid in explaination clarity?

KenCheung18 committed:
DG fix

TeoHaoZhi reviewed (commented):
Overall the DG looks clean and organised, however the diagrams are not rendered properly. Some commands could be better explained with reference to the sequence diagram.

KenCheung18 committed:
Merge branch 'master' into DG-fix

KenCheung18 committed:
Merge pull request #135 from KenCheung18/DG-fix

DG fix

Remy9926 commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,111 @@
+
+The architectural diagram of KaChinnnng is as follows:
+
+![ArchitectureDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/ArchitectureDiagram.png?raw=true)
```
The architecture diagram is a little complicated as it has a lot going on and each component has that figure in it, which is also distracting. Also, is the **Run** component supposed to have the **()**?

Remy9926 replied:
The architecture diagram is a little complicated as it has a lot going on and each component has that figure in it, which is also distracting. Also, is the Run component supposed to have the ()?

Remy9926 commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,111 @@
+Each of the categories inherit from a base Expense class which in turn
+inherits from the FinancialRecord class.
+
+![expense_class_diagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/expense_class_diagram.png?raw=true)
```
Do you have a key for the visibility of each method and attribute in your classes? It's not entirely clear which access modifiers each class' methods and attributes use.

Remy9926 commented on docs/DeveloperGuide.md:
```diff
@@ -2,12 +2,111 @@
+6. The saveExpenseToTextFile will create a new FileWriter and loop through the expenses list 
+7. It will get description, date, amount and type from the expenses list for every single expense record in the list and save the information into the text file
+
+![SaveToTxt_SequenceDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/SaveToTxt.png?raw=true)
```
In the diagram, it shows the fwe:FileWriter object being destroyed at the end of the program, but it gets instantiated again underneath the X? Is this how the program is intended to function?

Cheezeblokz commented on images/IncomeManager_seqdiagram.drawio.png:
What is being dispatched and returned for Ui object? Maybe it's possible to name the processes based on what it does, such as 'display added message'.

Cheezeblokz commented on images/IncomeManager_seqdiagram.drawio.png:
Should a dotted line be drawn from the end of IncomeManager's self-invocated activation bar to the primary activation bar?

Remy9926 commented on docs/DeveloperGuide.md:
````diff
@@ -2,12 +2,111 @@
+
+1. The user will first input the command to find the expense or income.
+```
+find /t <type> /cat [category] /de [description] /date [date]
````
It may be a little confusing as to why <type> is in <> while everything else is in []. This may be confusing to someone reading this, so maybe you can add a line explaining why it is different?

Cheezeblokz commented on images/expenseManager_sequence_diagram.png:
Should there be activation bars for Expense and ExchangeRateManager calls?

Cheezeblokz commented on images/GetFromTxt.png:
Could the standard Java utility classes such as IOException, Integer and Scanner be omitted to simplify the process and focus on the important operations?

Cheezeblokz reviewed (commented):
Generally clear description of process with ample use of diagrams.

bljhty committed:
PE-D bug fixing

bljhty committed:
no message

bljhty committed:
Merge pull request #171 from bljhty/Bryan-addincome_bugfixing

Bryan addincome bugfixing

bljhty committed:
edited code + finishes up the help command

bljhty committed:
edited expected output

bljhty committed:
Merge pull request #172 from bljhty/Bryan-moreBugFixes

edited code + finishes up the help command

bljhty committed:
no message

KenCheung18 committed:
UG fix

KenCheung18 committed:
Merge pull request #173 from KenCheung18/Bug_fix_on_UG

UG fix

KenCheung18 committed:
Change balance function to print only 2dp and without the scientific format

KenCheung18 committed:
style check fix

KenCheung18 committed:
Duke test fix

KenCheung18 committed:
Merge pull request #174 from KenCheung18/Bug_fix_on_balance_functioin

Change balance function to print only 2dp and without the scientific …

KenCheung18 committed:
change the website from google doc to github userguide

KenCheung18 committed:
change to userguide

KenCheung18 committed:
change expected

KenCheung18 committed:
Merge pull request #175 from KenCheung18/help_function_change

Help function change

limyuhching committed:
Debug edit income negative index error message

limyuhching committed:
Add showUpdateExchangeRateMessage in Ui and Update error messages

limyuhching committed:
Add space after commands that uses startsWith()

Updated clear commands to use equals() instead of startsWidth()

limyuhching committed:
Delete a comment on logger

limyuhching committed:
Add logger to UpdateExchangeRateCommand

limyuhching committed:
Merge pull request #178 from limyuhching/Debug.PE-D

Debug.PE-D.feature.ExchangeRate

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Merge pull request #179 from AY2324S1-CS2113-T18-3/limyuhching-Update-UG

Update UserGuide.md

limyuhching committed:
Merge remote-tracking branch 'origin/master' into Debug.PE-D

limyuhching committed:
Update sequence diagram

limyuhching committed:
Merge pull request #180 from limyuhching/master

Update sequence diagram

bljhty committed:
updated sequence diagram bug

Hongzhii committed:
change undefined to others for type category in expense

Hongzhii committed:
Merge branch 'rename-type-info-from-undefined-to-other'

* rename-type-info-from-undefined-to-other:
  change undefined to others for type category in expense

Hongzhii committed:
Merge pull request #182 from Hongzhii/master

change undefined to others for type category in expense

bljhty committed:
Merge branch 'master' into Bryan-updatePPP

Hongzhii committed:
no message

Hongzhii committed:
Merge branch 'update-ug-for-undefined-to-other-change'

* update-ug-for-undefined-to-other-change:
  no message

Hongzhii committed:
Merge pull request #183 from Hongzhii/master

update ug for undefined to other change

bljhty committed:
Merge pull request #181 from bljhty/Bryan-updatePPP

Bryan update ppp

bljhty committed:
no message

bljhty committed:
Merge pull request #184 from bljhty/Bryan-update_images

updated images in developers guide

limyuhching committed:
Update UpdateExchangeRate Sequence Diagram

limyuhching committed:
Merge branch 'master' of https://github.com/limyuhching/tp

limyuhching committed:
Merge pull request #185 from limyuhching/master

Update UpdateExchangeRateCommand Sequence Diagram

limyuhching committed:
Move welcome message after loadData()

And update "add income" to "add income " to invalidate "add incomeABCD"

limyuhching committed:
Merge pull request #186 from limyuhching/master

Move welcome message after loadData()

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Update UserGuide.md

limyuhching committed:
Update DeveloperGuide.md

KenCheung18 committed:
change the storage to deal with | and log

KenCheung18 committed:
Merge pull request #187 from KenCheung18/Storage-function-change

change the storage to deal with | and log

KenCheung18 committed:
Storage User guide and fix some bug

KenCheung18 committed:
UI

KenCheung18 committed:
UI

KenCheung18 committed:
change

KenCheung18 committed:
fix

KenCheung18 committed:
Merge pull request #188 from KenCheung18/Storage-function-change

Storage User guide and fix some bug

KenCheung18 committed:
Revert "Storage User guide and fix some bug"

KenCheung18 committed:
Merge pull request #189 from AY2324S1-CS2113-T18-3/revert-188-Storage-function-change

Revert "Storage User guide and fix some bug"

KenCheung18 committed:
add images

KenCheung18 committed:
Save

KenCheung18 committed:
Merge branch 'Add-image-to-UG' of https://github.com/KenCheung18/tp into Add-image-to-UG

KenCheung18 committed:
Merge pull request #190 from KenCheung18/Add-image-to-UG

add images

KenCheung18 committed:
Add image to the UG

KenCheung18 committed:
Merge pull request #191 from KenCheung18/Add-image-to-UG

Add image to the UG

KenCheung18 committed:
Change usageInstruction class

KenCheung18 committed:
Separate help function to different chunk for better UX

KenCheung18 committed:
UI test

KenCheung18 committed:
Merge pull request #192 from KenCheung18/Seperate-usage-instruction-to-different-function

Seperate usage instruction to different function

KenCheung18 committed:
Separate storage to two functions avoid function too long

KenCheung18 committed:
Change from list income to list incomes

KenCheung18 committed:
More JUnit test

KenCheung18 committed:
bug and style fix

KenCheung18 committed:
delete test case cuz not working in windows

KenCheung18 committed:
Style change

KenCheung18 committed:
fix

KenCheung18 committed:
Merge pull request #193 from KenCheung18/More-JUnit_test

More j unit test

Hongzhii committed:
no message

Hongzhii committed:
Merge branch 'update-uml-diagram'

* update-uml-diagram: (30 commits)
  no message
  fix
  Style change
  delete test case cuz not working in windows
  bug and style fix
  More JUnit test
  Change from list income to list incomes
  Separate storage to two functions avoid function too long
  UI test
  Separate help function to different chunk for better UX
  Change usageInstruction class
  Add image to the UG
  Save
  add images
  Revert "Storage User guide and fix some bug"
  fix
  change
  UI
  UI
  Storage User guide and fix some bug
  ...

Hongzhii committed:
Merge pull request #194 from Hongzhii/master

update uml diagram

KenCheung18 committed:
change diagram

KenCheung18 committed:
Merge pull request #195 from KenCheung18/Change-diagrams

change diagram

Hongzhii committed:
no message

Hongzhii committed:
Merge branch 'update-uml-diagram'

* update-uml-diagram:
  no message

Hongzhii committed:
Merge pull request #196 from Hongzhii/master

update uml diagram

limyuhching committed:
Create limyuhching.md

Hongzhii committed:
add eehongzhi.md

Hongzhii committed:
Merge pull request #197 from Hongzhii/master

add eehongzhi.md

Hongzhii committed:
udpate eehongzhi.cm

Hongzhii committed:
Merge pull request #198 from Hongzhii/master

udpate eehongzhi.cm

KenCheung18 committed:
Ka Yuen PPP

KenCheung18 committed:
change

KenCheung18 committed:
Merge pull request #199 from KenCheung18/KaYuen-PPP

Ka yuen ppp

Hongzhii committed:
change PPP title

Hongzhii committed:
Merge pull request #200 from Hongzhii/master

change PPP title

KenCheung18 committed:
Rename KaYuenCheung.md to KenCheung18.md

KenCheung18 committed:
Rename KenCheung18.md to kencheung18.md

bljhty committed:
edited PPP, user guide and DG

bljhty committed:
Merge pull request #201 from bljhty/update_ppp

edited PPP, user guide and DG

bljhty committed:
no message

bljhty committed:
Merge pull request #202 from bljhty/update_ppp

no message

bljhty committed:
no message

bljhty committed:
Merge pull request #203 from bljhty/Bryan-update_readme

no message

bljhty committed:
no message

bljhty committed:
Merge pull request #204 from bljhty/Bryan-update_readme

no message

Hongzhii committed:
Update README.md

Hongzhii committed:
Merge pull request #205 from Hongzhii/patch-1

Update README.md

Hongzhii committed:
Update README.md

Hongzhii committed:
Merge pull request #206 from Hongzhii/master

Update README.md

Hongzhii committed:
Update README.md

Hongzhii committed:
Update AboutUs.md

Hongzhii committed:
Update AboutUs.md

Hongzhii committed:
Merge pull request #207 from Hongzhii/master

update README.md

KenCheung18 committed:
Change the format of date

KenCheung18 committed:
Change help command to avoid bug

KenCheung18 committed:
Delete unused variable ui

KenCheung18 committed:
add Junit test for ListCommand

KenCheung18 committed:
Change List commands to ui.print for testing reason

KenCheung18 committed:
More Junit test

KenCheung18 committed:
Merge pull request #208 from KenCheung18/Typo-fix

Add Junit test, typo fix

KenCheung18 committed:
Change to userguide

KenCheung18 committed:
Merge pull request #209 from KenCheung18/update-user-guide

Change to userguide

KenCheung18 committed:
change user guide photos

KenCheung18 committed:
Merge pull request #210 from KenCheung18/user-guide-update-images

change user guide photos

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Typo in find parser

KenCheung18 committed:
UI fix for find command

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
change to Expected.txt

KenCheung18 committed:
import order change

KenCheung18 committed:
Merge pull request #211 from KenCheung18/fix-find-parser-bug

Fix find parser bug

KenCheung18 committed:
Update kencheung18.md

KenCheung18 committed:
minor bug fix

KenCheung18 committed:
change expected.txt

KenCheung18 committed:
Merge pull request #212 from KenCheung18/bug-fix

minor bug fix

limyuhching committed:
Update limyuhching.md

limyuhching committed:
Update DeveloperGuide.md

limyuhching committed:
Update DeveloperGuide.md

limyuhching committed:
Update DeveloperGuide.md

bljhty committed:
edited ug and dg

bljhty committed:
Merge pull request #213 from bljhty/final_edits

edited ug and dg

bljhty committed:
no message

bljhty committed:
Merge pull request #214 from bljhty/final_edits

no message

bljhty committed:
no message

bljhty committed:
Merge pull request #215 from bljhty/final_edits

no message

limyuhching committed:
Update DeveloperGuide.md

KenCheung18 committed:
Update DeveloperGuide.md

KenCheung18 committed:
Update DeveloperGuide.md

KenCheung18 committed:
Update DeveloperGuide.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Rename kencheung18.md to KenCheung18.md

KenCheung18 committed:
Update KenCheung18.md

KenCheung18 committed:
Update KenCheung18.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Update UserGuide.md

KenCheung18 committed:
Update KenCheung18.md

KenCheung18 committed:
Create kencheung18.md
